framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -0,0 +1,439 @@
+        } else {
+            printWriter.print(o.toString());
+        }
+    }
+
+    /** Prints an annotation in Java source format. */
+    private void printAnnotation(Annotation a) {
+        printWriter.print("@" + a.def().name.substring(a.def().name.lastIndexOf('.') + 1));
+        if (!a.fieldValues.isEmpty()) {
+            printWriter.print('(');
+            boolean first = true;
+            for (Map.Entry<String, Object> f : a.fieldValues.entrySet()) {
+                if (!first) {
+                    printWriter.print(',');
+                }
+                printWriter.print(f.getKey() + "=");
+                printValue(a.def().fieldTypes.get(f.getKey()), f.getValue());
+                first = false;
+            }
+            printWriter.print(')');
+        }
+    }
+
+    /**
+     * Prints all annotations in {@code annos}, separated by spaces. See {@link #printAnnotation}.
+     */
+    private void printAnnotations(Collection<? extends Annotation> annos) {
+        for (Annotation tla : annos) {
+            if (!tla.def.name.contains("+")) {
+                printWriter.print(' ');
+                printAnnotation(tla);
+            }
+        }
+    }
+
+    /** Finds and prints the annotations on the component type of an array, if there are any. */
+    private void printArrayComponentTypeAnnotation(ATypeElement e) {
+        for (Map.Entry<InnerTypeLocation, ATypeElement> ite : e.innerTypes.entrySet()) {
+            InnerTypeLocation loc = ite.getKey();
+            AElement it = ite.getValue();
+            if (loc.toString().contains("ARRAY")) {
+                printAnnotations(it.tlAnnotationsHere);
+                printWriter.append(' ');
+            }
+        }
+    }
+
+    /**
+     * Prints the stub representation of an AField, which represents any variable declaration. In
+     * practice, this should either be a field or a method parameters, since there should be no
+     * local variable declarations in a stub.
+     *
+     * <p>Takes two extra parameters beyond the AField: {@code fieldName} is the name to use for the
+     * declaration in the stub file. This doesn't matter for parameters, but must be correct for
+     * fields. {@code className} is the name of the enclosing class. This is only used for printing
+     * the type of an explicit receiver parameter (i.e. a parameter named "this").
+     */
+    private void printAField(AField aField, String fieldName, String className) {
+        String basetype;
+        if ("this".equals(fieldName)) {
+            basetype = className;
+        } else if (basetypes.containsKey(aField.type.description.toString())) {
+            basetype = basetypes.get(aField.type.description.toString()).toString();
+        } else {
+            basetype = "TODOTYPE";
+        }
+
+        // Annotations on arrays should be printed after the component type.
+        // Don't even bother checking for annotations on component types,
```

Comment:
Is this comment out-of-date? The code below this prints the annotations on the component type.

Reply:
It was out-of-date. I removed it.